dqc_us_rules/util/facts.py

```diff
@@ -175,15 +175,36 @@ def LegalEntityAxis_facts_by_member(facts):
     results = defaultdict(list)
     for fact in facts:
         legalDim = LEGALENTITYAXIS_DEFAULT
-        dims = [dim for dim in fact.context.segDimValues.values() if dim.isExplicit and dim.member is not None]
-        for dim in dims:
-            if dim.dimension.qname.localName == 'LegalEntityAxis':
-                legalDim = dim.member.qname.localName
-                break
-        results[legalDim].append(fact)
+        if fact_components_valid(fact):
+            dims = [dim for dim in fact.context.segDimValues.values() if dim.isExplicit and dim.member is not None]
+            for dim in dims:
+                if dim.dimension.qname.localName == 'LegalEntityAxis':
+                    legalDim = dim.member.qname.localName
```

Comment:
Not sure if a None check is necessary for defensiveness here, but if qname is none, this will blow up.